tests/unittests/config/test_cc_ca_certs.py

```diff
@@ -139,20 +144,23 @@ def test_remove_default_ca_certs(self):
         config = {"ca_certs": {"remove_defaults": True}}
 
         for distro_name in cc_ca_certs.distros:
-            self._mock_init()
+            self._mock_init(distro_name)
             cloud = get_cloud(distro_name)
             cc_ca_certs.handle(self.name, config, cloud, self.log, self.args)
 
             self.assertEqual(self.mock_add.call_count, 0)
             self.assertEqual(self.mock_update.call_count, 1)
-            self.assertEqual(self.mock_remove.call_count, 1)
+            if distro_name in ["fedora", "rhel"]:
+                self.assertEqual(self.mock_remove.call_count, 0)
+            else:
+                self.assertEqual(self.mock_remove.call_count, 1)
 
     def test_no_remove_defaults_if_false(self):
         """Test remove_defaults is not called when config value is False."""
         config = {"ca_certs": {"remove_defaults": False}}
 
         for distro_name in cc_ca_certs.distros:
-            self._mock_init()
+            self._mock_init(distro_name)
             cloud = get_cloud(distro_name)
```

Comment:
and so forth

Reply:
again, this is after _mock_init is called where the mock is defined.